android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/filters.kt

```diff
@@ -0,0 +1,493 @@
+						lowerTextFieldError =
+							!verifyNewBoundVal(newVal = lowerBoundText, upperBound = upperboundText)
+					},
+					isError = lowerTextFieldError,
+					placeholderString = stringResource(id = R.string.minimum),
+				)
+				Divider(
+					modifier = Modifier.width(width = dimensionResource(id = R.dimen.m)),
+					color = secondaryTextColor,
+					thickness = dimensionResource(id = R.dimen.xxxxs)
+				)
+				TextFieldWithErrorIndication(
+					value = upperboundText,
+					onValueChange = {
+						upperboundText = it
+						upperTextFieldError =
+							!verifyNewBoundVal(newVal = upperboundText, lowerBound = lowerBoundText)
+					},
+					isError = upperTextFieldError,
+					placeholderString = stringResource(id = R.string.maximum),
+
+					)
+			}
+		},
+		clearAction = {
+			lowerBoundText = ""
+			upperboundText = ""
+			lowerTextFieldError = false
+			upperTextFieldError = false
+		},
+		updateFilterAndClose = {
+			if (!lowerTextFieldError && !upperTextFieldError) {
+				updatePriceFilter(
+					HomeListUiState.PriceRange(
+						lowerBoundText.toIntOrNull(),
+						upperboundText.toIntOrNull()
+					)
+				)
+				closeAction()
+
+			}
+		},
+		revertInput = {
+			lowerBoundText =
+				if (currentPriceRange.lowerBound == null) "" else currentPriceRange.lowerBound.toString()
+			upperboundText =
+				if (currentPriceRange.upperBound == null) "" else currentPriceRange.upperBound.toString()
+			lowerTextFieldError = false
+			upperTextFieldError = false
+
+		}
+	)
+
+}
+
+@Composable
+fun BasicFilterLayout(
+	modifier: Modifier = Modifier,
+	@StringRes titleId: Int,
+	closeAction: () -> Unit,
+	content: @Composable (() -> Unit)? = null,
+	updateFilterAndClose: () -> Unit,
+	clearAction: () -> Unit,
+	revertInput: () -> Unit,
+) {
+	Scaffold(
+		modifier = modifier
+			.fillMaxWidth(1.0f),
+		containerColor = Color.White,
+		topBar = {
+			Row(
+				modifier = Modifier
+					.fillMaxWidth(1.0f)
+					.height(dimensionResource(id = R.dimen.xl))
+					.padding(
+						dimensionResource(id = R.dimen.m),
+						dimensionResource(id = R.dimen.xxs)
+					),
+				horizontalArrangement = Arrangement.SpaceBetween
+			)
+			{
+				Icon(
+					modifier = Modifier/**/.clickable {
+						closeAction()
+						revertInput()
+					},
+					painter = painterResource(
+						id = R.drawable.close_filled_black_24
+					),
+					contentDescription = stringResource(id = R.string.close),
+				)
+				Text(
+					stringResource(id = titleId),
+					style = MaterialTheme.typography.displayMedium
+				)
+				Spacer(modifier = Modifier.width(dimensionResource(id = R.dimen.m)))
+
+			}
+		},
+		bottomBar = {
+			Column {
+				Divider(
+					color = textFieldBackgroundColor,
+					modifier = Modifier
+						.height(dimensionResource(id = R.dimen.xxxxs))
+						.fillMaxWidth()
+				)
+				Row(
+					modifier = Modifier
+						.fillMaxWidth(1.0f)
+						.padding(
+							dimensionResource(id = R.dimen.m),
+							dimensionResource(id = R.dimen.xxs)
```

Comment:
Add trailing comma.
Also, can you make these named parameters (makes the code easier to read).